chainercv/links/model/pspnet/pspnet.py

```diff
@@ -0,0 +1,476 @@
+            self.cbr1 = ConvBNReLU(
+                in_ch, mid_ch, 1, 1 if mid_stride else stride, 0)
+            if dilate:
+                self.cbr2 = ConvBNReLU(mid_ch, mid_ch, 3, 1, dilate, dilate)
+            else:
+                self.cbr2 = ConvBNReLU(
+                    mid_ch, mid_ch, 3, stride if mid_stride else 1, 1)
+            self.cbr3 = ConvBNReLU(mid_ch, out_ch, 1, 1, 0)
+            self.cbr4 = ConvBNReLU(in_ch, out_ch, 1, stride, 0)
+
+    def __call__(self, x):
+        h = self.cbr1(x)
+        h = self.cbr2(h)
+        h1 = self.cbr3(h, relu=False)
+        h2 = self.cbr4(x, relu=False)
+        return F.relu(h1 + h2)
+
+
+class BottleneckIdentity(chainer.Chain):
+
+    def __init__(self, in_ch, mid_ch, dilate=False):
+        super(BottleneckIdentity, self).__init__()
+        with self.init_scope():
+            self.cbr1 = ConvBNReLU(in_ch, mid_ch, 1, 1, 0)
+            if dilate:
+                self.cbr2 = ConvBNReLU(mid_ch, mid_ch, 3, 1, dilate, dilate)
+            else:
+                self.cbr2 = ConvBNReLU(mid_ch, mid_ch, 3, 1, 1)
+            self.cbr3 = ConvBNReLU(mid_ch, in_ch, 1, 1, 0)
+
+    def __call__(self, x):
+        h = self.cbr1(x)
+        h = self.cbr2(h)
+        h = self.cbr3(h, relu=False)
+        return F.relu(h + x)
+
+
+class ResBlock(chainer.ChainList):
+
+    def __init__(self, n_layer, in_ch, mid_ch, out_ch, stride):
+        super(ResBlock, self).__init__()
+        self.add_link(BottleneckConv(in_ch, mid_ch, out_ch, stride))
+        for _ in six.moves.xrange(1, n_layer):
+            self.add_link(BottleneckIdentity(out_ch, mid_ch))
+
+    def __call__(self, x):
+        for f in self:
+            x = f(x)
+        return x
+
+
+class DilatedResBlock(chainer.ChainList):
+
+    def __init__(self, n_layer, in_ch, mid_ch, out_ch, dilate):
+        super(DilatedResBlock, self).__init__()
+        self.add_link(BottleneckConv(in_ch, mid_ch, out_ch, 1, dilate))
+        for _ in six.moves.xrange(1, n_layer):
+            self.add_link(BottleneckIdentity(out_ch, mid_ch, dilate))
+
+    def __call__(self, x):
+        for f in self:
+            x = f(x)
+        return x
+
+
+class DilatedFCN(chainer.Chain):
+
+    def __init__(self, n_blocks):
+        super(DilatedFCN, self).__init__()
+        with self.init_scope():
+            self.cbr1_1 = ConvBNReLU(None, 64, 3, 2, 1)
+            self.cbr1_2 = ConvBNReLU(64, 64, 3, 1, 1)
+            self.cbr1_3 = ConvBNReLU(64, 128, 3, 1, 1)
+            self.res2 = ResBlock(n_blocks[0], 128, 64, 256, 1)
+            self.res3 = ResBlock(n_blocks[1], 256, 128, 512, 2)
+            self.res4 = DilatedResBlock(n_blocks[2], 512, 256, 1024, 2)
+            self.res5 = DilatedResBlock(n_blocks[3], 1024, 512, 2048, 4)
+
+    def __call__(self, x):
+        h = self.cbr1_3(self.cbr1_2(self.cbr1_1(x)))  # 1/2
+        h = F.max_pooling_2d(h, 3, 2, 1)  # 1/4
+        h = self.res2(h)
+        h = self.res3(h)  # 1/8
+        if chainer.config.train:
```

Comment:
Why not always return `h1` and `h2`?

Reply:
`h1` is only for calculating auxiliary loss. That is completely useless during inference.

Reply:
`predict` should be called during inference and not `__call__`. There is not much inconvenience returning it.

Reply:
In general, I want to avoid using a global variable to configure behavior if possible.

Reply:
OK, I changed this part to return both `h1` and `h2` always.